tests/engine/artifact_filters.py

```diff
@@ -304,7 +340,134 @@ def testBuildFindSpecsFromFileSourcePath(self):
     self.assertEqual(
         find_specs[15]._location_segments, expected_location_segments)
 
-  # TODO: add tests for _BuildFindSpecsFromRegistrySourceKey
+    # Test that paths are added to artifacts trie.
+    self.assertIn(os.sep, test_filter_file.artifacts_trie.root.children)
+    path_trie_node = test_filter_file.artifacts_trie.root.children[os.sep]
+    self.assertEqual(
+        path_trie_node.artifacts_names, [])
+    self.assertEqual(len(path_trie_node.children), 5)
+    self.assertIn('Windows', path_trie_node.children)
+    self.assertIn('test_data', path_trie_node.children)
+    self.assertIn('home', path_trie_node.children)
+    self.assertIn('Users', path_trie_node.children)
+    self.assertIn('homes', path_trie_node.children)
+
+    self.assertEqual(
+        path_trie_node.children['Windows'].artifacts_names, [])
+    self.assertIn(
+        'test_data', path_trie_node.children['Windows'].children)
+
+    self.assertEqual(
```

Comment:
Instead of repeatedly accessing members why not use a local variable for referencing the children?